### PR TITLE
Adds MAC cannon

### PR DESCRIPTION
**Pull request by Vivalas**
(Don't know what the hell is up with mapmerge. I screwed up hardcore and then tried to start over and re-did the changes at least 3 times, with no change in the massive amount of diffs.)

Adds the MAC cannon (Magnetic Accelerator Cannon) which might be renamed.

The cannon can fire both people and shells. When loading something into the cannon, the breech must be opened first, and then the shell loaded. Shells must be armed with a multitool first before launching, otherwise they won't do any damage. However, armed shells will explode when ejected out of the breech.

There is no limit to the amount of things you can load before firing but loading a non-shell and a shell at the same time will result in an explosion.

When fired, the cannon takes 10 seconds to recharge. After firing, the breech will need to be opened to load another shell, which will also eject the spent casing.

Munitions

image

### CODE REVIEW

**Comment by monster860** [comment]
You're fine, @CrAzYPiLoT-SS13 forgot to mapmerge.

**Comment by CrAzYPiLoT-SS13** [comment]
Sorry.

File: code/game/machinery/mac_cannon.dm [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		spawn(0) M.close()
+
+
+/obj/machinery/mac_barrel/proc/fire_sound()
+	playsound(loc, 'sound/weapons/flashbang.ogg', 100, 1)
+	for(var/mob/living/carbon/human/H in range(5,src))
+		if(H.check_ear_prot()) continue
+		if(!istype(H.loc.loc,/area/shuttle/ftl/munitions)) continue
+
+		H.show_message("<span class='warning'>BANG</span>", 2)
```
Comment (CrAzYPiLoT-SS13) [inline code/game/machinery/mac_cannon.dm]: Memes.

File: code/game/machinery/mac_cannon.dm [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	anchored = 1
+
+/obj/machinery/mac_breech/New()
+	..()
+	var/obj/item/weapon/circuitboard/machine/B = new /obj/item/weapon/circuitboard/machine/mac_cannon(null)
+	B.apply_default_parts(src)
+	RefreshParts()
+
+
+/obj/item/weapon/circuitboard/machine/mac_cannon
```
Comment (CrAzYPiLoT-SS13) [inline code/game/machinery/mac_cannon.dm]: Isn't there a separate file for defining machine circuitboards or am I thinking of computers?

File: code/game/machinery/mac_cannon.dm [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	build_path = /obj/machinery/mac_breech
+	origin_tech = "programming=3;powerstorage=4;combat=4"
+	req_components = list(
+							/obj/item/weapon/stock_parts/scanning_module = 1,
+							/obj/item/weapon/stock_parts/capacitor = 1,
+							/obj/item/weapon/stock_parts/console_screen = 1,)
+
+
+/obj/machinery/mac_breech/process()
+	var/is_charged = charge_process >= 100 ? 1 : 0
```
Comment (CrAzYPiLoT-SS13) [inline code/game/machinery/mac_cannon.dm]: Why would you use ternary operators in a non-string ;_;

File: code/game/objects/items/devices/radio/headset.dm [hunk 4]
```diff
@@ -163,6 +163,10 @@
 	name = "\proper the head of personnel's headset"
 	desc = "The headset of the guy who will one day be captain. \nChannels are as follows: :u - supply, :v - service, :c - command."
 	icon_state = "com_headset"
+	flags = EARBANGPROTECT
```
Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/devices/radio/headset.dm]: Note to self: Refactor earmufflers

File: code/game/objects/structures/cannon_shell.dm [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+	var/armed = 0
+
+	var/damage = 5
+
+
+/obj/structure/shell/attackby(obj/item/C,mob/user)
+	if(istype(C,/obj/item/device/multitool))
+		playsound(loc,'sound/weapons/empty.ogg',50,0)
+		user.visible_message("<span class=notice>[user] toggles the arming bolts on [src].</span>","<span class=notice>You toggle the arming bolts on [src]</span>")
```
Comment (CrAzYPiLoT-SS13) [inline code/game/objects/structures/cannon_shell.dm]: Maybe a better one would be "[user] toggles the arming mechanism on [src]."

File: code/game/machinery/mac_cannon.dm (hunk 2 above)
Comment (monster860) [inline code/game/machinery/mac_cannon.dm]: Nope, on /tg/ the boards are defined next to the machine

File: code/game/machinery/mac_cannon.dm (hunk 3 above)
Comment (monster860) [inline code/game/machinery/mac_cannon.dm]: You could just do `var/is_charged = charge_process >= 100`

File: code/game/machinery/mac_cannon.dm (hunk 1 above)
Comment (Vivalas) [inline code/game/machinery/mac_cannon.dm]: This is probably the part I hate the most. It feels so cheezy copying flashbang code.

File: code/game/machinery/mac_cannon.dm (hunk 2 above)
Comment (Vivalas) [inline code/game/machinery/mac_cannon.dm]: Speaking of which I forgot to define the components and shit for the barrel.

**Comment by Vivalas** [comment]
I was gonna use power cells to charge it but decided I wanted to make it a flat 10 seconds, due to the length of the charge sound.

**Comment by Vivalas** [comment]
Added picture of munitions.

**Pull request merged**